File: 1894-merge-strings-alternately/merge-strings-alternately.go
func mergeAlternately(word1 string, word2 string) string {
    maxLen := len(word1)
    if len(word2) > maxLen {
        maxLen = len(word2)
    }

    result := make([]byte, 0, len(word1) + len(word2))

    for i := 0; i < maxLen; i++ {
        if i < len(word1) {
            result = append(result, word1[i])
        }

        if i < len(word2) {
            result = append(result, word2[i])
        }
    }

    return string(result)
} 

File: 1894-merge-strings-alternately/merge-strings-alternately.rs
impl Solution {
    pub fn merge_alternately(word1: String, word2: String) -> String {
        let mut result = String::with_capacity(word1.len() + word2.len());
        let (word1_chars, word2_chars) = (word1.chars().collect::<Vec<_>>(), word2.chars().collect::<Vec<_>>());
        let max_len = word1.len().max(word2.len());

        for i in 0..max_len {
            if i < word1.len() {
                result.push(word1_chars[i]);
            }
            if i < word2.len() {
                result.push(word2_chars[i]);
            }
        }

        result
    }
}